cmd/clusterawsadm/api/bootstrap/v1alpha1/types.go

```diff
@@ -192,6 +192,10 @@ type AWSIAMConfigurationSpec struct {
 	// will generate AWS Secrets Manager policies instead.
 	// +kubebuilder:validation:Enum=secrets-manager;ssm-parameter-store
 	SecureSecretsBackends []infrav1.SecretBackend `json:"secureSecretBackends,omitempty"`
+
+	// SecureSecretsBackend, when set to true, will add controller nodes permissions to
+	// create S3 Buckets for workload clusters.
+	S3Bucket bool `json:"s3Bucket,omitempty"`
```

Comment:
Change this to a struct so we can have the following YAML

```
S3Buckets:
  Enabled: true
  # can add now or later
  Prefix: ""
```

This is so that the user can control which S3 buckets the provider is allowed to create (see next comment)

Reply:
Hmm, perhaps prefix should have some default value then, like `capz` or whatever default prefix is used around. This way, by default access is limited to CAPZ-specific buckets only. Any idea what prefix we could use?

Reply:
No prefix for now, so marking as resolved.